doc/specs/stdlib_logger.md

```diff
@@ -0,0 +1,965 @@
+
+### `add_log_file` - open a file and add its unit to `self % log_units`
+
+#### Status
+
+Experimental
+
+#### Description
+
+Opens a formatted, sequential access, output file, `filename` using
+`newunit` and adds the resulting unit number to the logger's
+`log_units` array.
+
+#### Syntax
+
+`call [[stdlib_logger(module):self % add_log_file(interface)]]( filename [, unit, action, position, status, stat ] )`
+
+#### Class
+
+Subroutine
+
+#### Arguments
+
+`self`: shall be a scalar variable of type `logger_type`. It is an
+`intent(inout)` argument. It shall be the logger to add the file to its `log_units`.
+
+`filename`: shall be a scalar default character expression. It is
+an `intent(in)` argument. It shall be the name of the file to be opened.
+
+`unit` (optional): shall be a scalar default integer variable. It is an
+`intent(out)` argument. It will be the unit number returned by the
+`newunit` specifier of the `open` statement for `filename`.
+
+`action` (optional): shall be a scalar default character
+  expression. It is an `intent(in)` argument. It shall be the `action`
+  specifier of the `open` statement and must have one of the values
+  `'WRITE'` or `'READWRITE'`. It has the default value of `'WRITE'`.
+
+`position` (optional): shall be a scalar default character
+  expression. It is an `intent(in)` argument. It shall be the
+  `position` specifier of the `open` statement and must have one of
+  the values `'ASIS'`, `'REWIND'`, or `'APPEND'`. It has the default
+  value of `'REWIND'`.
+
+`status` (optional): shall be a scalar default character
+  expression. It is an `intent(in)` argument. It shall be the
+  `status` specifier of the `open` statement and must have one of
+  the values `'OLD'`, `'NEW'`, `'REPLACE'`, or `'UNKNOWN'`. It has the
+  default value of `'REPLACE'`.
+
+`stat` (optional): shall be a scalar default integer variable. It
+  is an `intent(out)` argument. If present, on return it will have the
+  value `success` if `filename` could be opened, the value
+  `read_only_error` if the `action` specifier is `"READ"`, or the value
+  `open_failure` if `filename` could not be opened. If absent and `filename`
+  could not be opened then processing will stop with an informative message as the stop code.
+
+#### Example
+
+```fortran
+program demo_global_logger
+    use stdlib_logger, global => global_logger
+    ...
+    integer :: unit, stat
+    ...
+    call global % add_log_file( 'error_log.txt', unit, &
+        position='asis', stat=stat )
+    if ( stat /= success ) then
+         error stop 'Unable to open "error_log.txt".'
+    end if
+     ...
+end program demo_global_logger
+```
+
+### `add_log_unit` - add a unit to the array `self % log_units`
+
+#### Status
+
+Experimental
+
+#### Description
+
+Adds `unit` to the array of `self % log_units`. The `unit` shall
+be the unit number for an opened, sequential, formatted file with an
+`action` specifier of `'WRITE'` or `'READWRITE'`. Failure of `unit` to meet
```

Comment:
```suggestion
`action` specifier of `'WRITE'` or `'READWRITE'`. If `unit` fail to meet
```

Reply:
Should it be:
"If `unit` fails to meet", instead of "fail"?

Reply:
This suggestion is outdated (Yes, it should be. "it fails" ~and I failed it~) xD